CAPI/API/src/logic.cpp

```diff
@@ -384,6 +425,9 @@ void Logic::LoadBuffer(std::shared_ptr<Protobuf::MessageToClient> pm2c)
 
 void Logic::PlayerWrapper(std::function<void()> player)
 {
+#ifdef __linux__
+    pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+#endif
     {
         std::unique_lock<std::mutex> lock(mtx_ai);
         cv_ai.wait(lock, [this]() {return AI_start; }); // 突然发现此处不能返回atomic_bool类型，所以THUAI4才会搞出另一个控制AI是否启动的标志值
```

Comment:
看到这里的注释顺便说一下，这里倒不是不能原子，只需要 `cv_ai.wait(lock, [this]() -> bool { return AI_start; });` 或者 `cv_ai.wait(lock, [this]() { return static_case<bool>(AI_start); });` 就可以了。不过这样做没什么意义，用普通的 `bool` 就可以了。如果原子变量支持 `wait` 操作，那么也不需要 `mutex` 和条件变量了（对原子变量的 `wait` 操作需要 C++20）。
```suggestion
        cv_ai.wait(lock, [this]() { return AI_start; });
```